src/mgmt_conn.c

```diff
@@ -1569,17 +1576,15 @@ process_message(uzfs_mgmt_conn_t *conn)
 		}
 		strlcpy(zvol_name, payload, payload_size);
 		zvol_name[payload_size] = '\0';
-		if (hdrp->opcode != ZVOL_OPCODE_SNAP_LIST) {
-			snap = strchr(zvol_name, '@');
-			if (snap == NULL) {
-				LOG_ERR("Invalid snapshot name: %s",
-				    zvol_name);
-				rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED,
-				    hdrp);
-				break;
-			}
+		snap = strchr(zvol_name, '@');
+		if (snap == NULL && hdrp->opcode != ZVOL_OPCODE_SNAP_LIST) {
+			LOG_ERR("Invalid snapshot name: %s", zvol_name);
+			rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED, hdrp);
+			break;
+		} else if (snap != NULL) {
```

Comment:
I understand this code is correct.. but, for better readability and maintainability, lets decouple the cases here by removing 'else' and by just having 'if'

Reply:
done